File: app/src/main/java/com/merino/ddfilms/model/MovieLists.java
package com.merino.ddfilms.model;

import java.util.List;

import lombok.Data;

@Data
public class MovieLists {
    String id;
    String name;
    String description;
    List<String> userID;
    List<Movie> movies;

    // Denormalized fields
    List<String> coverPreviews;
    List<String> memberAvatarsPreview;
    String lastUpdated;
    int moviesCount;
}

File: app/src/main/java/com/merino/ddfilms/model/MovieLists.kt
package com.merino.ddfilms.model

data class MovieLists(
    var id: String? = null,
    var name: String? = null,
    var description: String? = null,
    var userID: List<String>? = null,
    var movies: List<Movie>? = null,
    var coverPreviews: List<String>? = null,
    var memberAvatarsPreview: List<String>? = null,
    var lastUpdated: String? = null,
    var moviesCount: Int = 0
)
